### ISSUE TITLE
DULPreflightChecklistManager Returns Inaccurate ReadyToFly State

### ISSUE THREAD

**Issue by rplumbo** [dji-sdk/Mobile-SDK-iOS#222]
The DULPreflightChecklistManager incorrectly returns ReadyToFly = true when the batteries are too cold to actually launch.

**Declaration:**
`let checklistManager: DULPreflightChecklistManager = DULPreflightChecklistManager.default()`

**Call to start checking:**
`checklistManager.startCheckingList()`

**Query:**
```
if checklistManager.isReadyToFly {
DJISDKManager.missionControl?.startTimeline()
}
```

Steps to Reproduce:
1. Cool battery to below 10.0C.
2. Insert battery, run code above.
3. Observe aircraft does not take off.
4. Open DJI GO.
5. Attempt to Launch using DJI GO.
6. Observer error that battery is too cold to launch.

Steps to Reproduce:
1. Warm battery above 15.0C.
2. Insert battery, run code above.
3. Observe aircraft takes off.

Using the timeline listener code below, you can observe the timeline iterate three times through a sequence of .Started, .Progressed, .Finished.
`DJISDKManager.missionControl()?.addListener(self, toTimelineProgressWith: { (timelineEvent, timelineElement, error, info) in...`

Notably, the listener provides no error as to why the aircraft did not take off. Only by using DJIDiagnostics or other methods are we able to discern the reason.

Further, the DULPreflightChecklistManager also incorrectly returns ReadyToFly = true when IMUs are preheating. This can be observed by cooling the aircraft to roughly 0C and following the same steps above.

**However, in this case,** the specific DULPreflightChecklistItem isReadyToFly property for IMU will return false, but the total .isReadyToFly state of the checklistManager will remain true.

In short, I would like a single, rolled up object I can query for a ReadyToFly state. I don't want to individually validate each flight system. This appears to be the intent of the DULPreflightChecklistManager, which is fantastic, but it seems to have some inconsistencies either in our understanding of usage, or its performance.

Thank you!

SDK: 4.4
XCode 9.2
Phantom 3 Professional, Inspire 1
iOS 11.2
iPhone X, iPad Air
